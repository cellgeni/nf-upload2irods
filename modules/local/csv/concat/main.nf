process CSV_CONCAT {
    tag "Concatenating CSV files"
    container 'docker://quay.io/cellgeni/metacells-python:latest'

    input:
    tuple val(meta), path(csv_files, name: "input/*.csv")

    output:
    tuple val(meta), path("*.csv"), emit: csv
    path "versions.yml", emit: versions

    script:
    def prefix = task.ext.prefix ?: "concatenated"
    def args = task.ext.args ?: ""
    """
    concat.py --input ${csv_files} --prefix "${prefix}.csv" ${args}
    
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | awk '{print \$2}')
        pandas: \$( python -c "import pandas; print(pandas.__version__)" )
    END_VERSIONS
    """
}
