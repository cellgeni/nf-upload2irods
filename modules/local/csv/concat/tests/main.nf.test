nextflow_process {
    name "Test Process: CSV_CONCAT"
    script "../main.nf"
    process "CSV_CONCAT"

    tag "modules"
    tag "csv"
    tag "csv/concat"

    test("Concatenate multiple CSV files") {
        tag "basic"

        when {
            process {
                """
                input[0] = [
                    [id: "test_concat"],
                    [
                        file(params.test_data_base + "csv/concat/file1.csv"),
                        file(params.test_data_base + "csv/concat/file2.csv"),
                        file(params.test_data_base + "csv/concat/file3.csv")
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.versions },
                { assert process.out.csv },
                { assert process.out.json },
                { assert path( process.out.csv.get(0).get(1) ).md5 == path( "tests/data/csv/concat/results.csv" ).md5 },
                { assert snapshot(process.out).match() }
            )
        }
    }
}