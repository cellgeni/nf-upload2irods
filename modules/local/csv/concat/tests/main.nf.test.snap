{
    "Concatenate multiple CSV files": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_concat"
                        },
                        "metadata.csv:md5,969e0c9fee0ad4156eb2f6ac2e7ff815"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_concat"
                        },
                        "metadata.json:md5,e9573a3c0f88048e12a5655a4bc80d5a"
                    ]
                ],
                "2": [
                    "versions.yml:md5,efc3533460e957d1714d02d8dd758890"
                ],
                "csv": [
                    [
                        {
                            "id": "test_concat"
                        },
                        "metadata.csv:md5,969e0c9fee0ad4156eb2f6ac2e7ff815"
                    ]
                ],
                "json": [
                    [
                        {
                            "id": "test_concat"
                        },
                        "metadata.json:md5,e9573a3c0f88048e12a5655a4bc80d5a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,efc3533460e957d1714d02d8dd758890"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.4"
        },
        "timestamp": "2025-10-24T17:20:52.724355864"
    }
}